go/libkb/api_stub.go

```diff
@@ -5,31 +5,146 @@
 
 package libkb
 
+import "fmt"
+
 type StubAPIEngine struct {
 	*ExternalAPIEngine
+	mocking       bool
+	expectingJSON map[string]*ExternalAPIRes
+	expectingHTML map[string]*ExternalHTMLRes
+	expectingText map[string]*ExternalTextRes
+	calls         []stubAPIEngineCallRecord
+}
+
+type stubAPIEngineCallRecord struct {
+	kind     XAPIResType
+	endpoint string
 }
 
 func NewStubAPIEngine() *StubAPIEngine {
 	return &StubAPIEngine{
 		ExternalAPIEngine: &ExternalAPIEngine{BaseAPIEngine{clients: make(map[int]*Client)}},
+		mocking:           false,
+		expectingJSON:     make(map[string]*ExternalAPIRes),
+		expectingHTML:     make(map[string]*ExternalHTMLRes),
+		expectingText:     make(map[string]*ExternalTextRes),
+		calls:             make([]stubAPIEngineCallRecord, 0),
 	}
 }
 
 func (e *StubAPIEngine) Get(arg APIArg) (res *ExternalAPIRes, err error) {
+	if e.mocking {
```

Comment:
in what case would this StubAPI be linked to from `G` but we would want mocking off?